lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1674,9 +1674,6 @@ void RemoveUnusedInternalGlobalVariable(Module &M) {
         Value *User = *(UserIt++);
         if (Instruction *I = dyn_cast<Instruction>(User)) {
           I->eraseFromParent();
-        } else {
-          ConstantExpr *CE = cast<ConstantExpr>(User);
-          CE->dropAllReferences();
```

Comment:
Could there still be constant users which are used by instructions that would prevent `GV->eraseFromParent()` from succeeding?

Reply:
Oh maybe it's not possible because it's doing:
```
if (isa<ConstantExpr>(user) && user->user_empty())
  continue;
```
Then, if it's a constant user that's not `user_empty()`, it will fall through to `onlyStoreUse = false` and `break;`, preventing it from going down this code path.